src/components/combobox/combobox.tsx

```diff
@@ -505,7 +509,7 @@ export class Combobox implements LabelableComponent, FormComponent, InteractiveC
       return;
     }
 
-    if (this.allowCustomValues && this.text) {
+    if (this.allowCustomValues && this.text?.trim().length) {
```

Comment:
I _think_ there's no need to use conditional chaining here since `text` should be guaranteed (it's initialized/reset to "" and is always updated from an `<input>`'s value).